component/src/main/java/org/wso2/carbon/identity/authenticator/office365/Office365Authenticator.java

```diff
@@ -0,0 +1,339 @@
+        }
+        return oAuthResponse;
+    }
+
+    /**
+     * Get the access request
+     *
+     * @param tokenEndPoint Office365 authorize endpoint URL.
+     * @param clientId      The client ID of the client application that is registered in Azure AD.
+     * @param code          Authorization code to be used to obtain the access token.
+     * @param clientSecret  The value of the key that contains the client password.
+     * @param callbackUrl   Specifies the reply URL of the application.
+     * @return Response OAuthClientRequest.
+     * @throws AuthenticationFailedException
+     */
+    private OAuthClientRequest getAccessRequest(String tokenEndPoint, String clientId, String code, String clientSecret,
+                                                String callbackUrl) throws AuthenticationFailedException {
+        OAuthClientRequest accessRequest;
+        try {
+            accessRequest = OAuthClientRequest.tokenLocation(tokenEndPoint)
+                    .setGrantType(GrantType.AUTHORIZATION_CODE)
+                    .setCode(code)
+                    .setRedirectURI(callbackUrl)
+                    .setClientId(clientId)
+                    .setClientSecret(clientSecret)
+                    .buildBodyMessage();
+        } catch (OAuthSystemException e) {
+            throw new AuthenticationFailedException("Exception while building request for request access token " +
+                    e.getMessage(), e);
+        }
+        return accessRequest;
+    }
+
+    /**
+     * Request user claims from user info endpoint.
+     *
+     * @param url         User info endpoint.
+     * @param accessToken Access token.
+     * @return Response string.
+     * @throws IOException
+     */
+    @Override
+    protected String sendRequest(String url, String accessToken) throws IOException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Claim URL: " + url);
+        }
+        if (url == null) {
+            return StringUtils.EMPTY;
+        }
+        URL obj = new URL(url);
+        HttpURLConnection urlConnection = (HttpURLConnection) obj.openConnection();
+        urlConnection.setRequestMethod(Office365AuthenticatorConstants.HTTP_GET);
+        urlConnection.setRequestProperty(Office365AuthenticatorConstants.HEADER_ACCEPT,
+                Office365AuthenticatorConstants.ODATA_ACCEPT_HEADER);
+        urlConnection.setRequestProperty(Office365AuthenticatorConstants.Authorization,
+                Office365AuthenticatorConstants.BEARER + accessToken);
+        BufferedReader reader = new BufferedReader(new InputStreamReader(urlConnection.getInputStream()));
+        StringBuilder builder = new StringBuilder();
+        String inputLine = reader.readLine();
+        while (inputLine != null) {
+            builder.append(inputLine).append("\n");
+            inputLine = reader.readLine();
+        }
+        reader.close();
```

Comment:
Shouldn't this be in the finally block ? what if the the read fails. There could be a potential leak

Reply:
Fixed in https://github.com/wso2-extensions/identity-outbound-auth-office365/pull/3